Include/internal/pycore_suggestions.h

```diff
@@ -0,0 +1,12 @@
+#include "Python.h"
+
+#ifndef Py_INTERNAL_SUGGESTIONS_H
+#define Py_INTERNAL_SUGGESTIONS_H
+
+#ifndef Py_BUILD_CORE
+#  error "this header requires Py_BUILD_CORE define"
+#endif
+
+PyObject* _Py_Offer_Suggestions(PyObject* exception);
```

Comment:
You can put it in Include/internal/pycore_pyerrors.h rather than adding an header file just for a single function declaration.